networkx/generators/tests/test_random_graphs.py

```diff
@@ -308,10 +308,19 @@ def root(u, w, r):
             return r / c + w
 
         c = 1
-        graph = nx.random_kernel_graph(1000, integral, root)
         graph = nx.random_kernel_graph(1000, integral, root, seed=42)
         assert len(graph) == 1000
 
+    def test_random_kernel_graph_default_root(self):
+        pytest.importorskip("scipy")
```

Comment:
It's not immediately obvious why this is necessary, so a comment might save a lookup

```suggestion
        """When kernel_root is None, sp.optimize.brentq is used to 
        construct the default kernel"""
        pytest.importorskip("scipy")
```